### PR TITLE
All Code Docs Edit Page

### PR DESCRIPTION
**Pull request by bethanyaconnor**
Adds a page to see all code docs, with links to editing, cloning to another environment, and the ability to destroy code docs. See mock in [the spec](https://docs.google.com/document/d/1CuMG8vlECM0xLs8kq5hocUS5ToyPWipjTGU841W2wvs/edit#bookmark=id.dwxvqefs3lom). This PR adds the table for programming environments, the component that holds both that and the table for programming expressions (added in #45391), and the route for the page (which is currently /programming_expressions but that might end up changing).

https://user-images.githubusercontent.com/46464143/162782342-64bd6458-cec1-4a6c-b5fb-050ebe6d9918.mov

 

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/lib/levelbuilder/code-docs-editor/AllCodeDocs.jsx [hunk 1]
```diff
@@ -0,0 +1,104 @@
+import React, {useState} from 'react';
+import PropTypes from 'prop-types';
+import ProgrammingEnvironmentsTable from './ProgrammingEnvironmentsTable';
+import ProgrammingExpressionsTable from './ProgrammingExpressionsTable';
+import {TextLink} from '@dsco_/link';
+
+export default function AllCodeDocs({programmingEnvironments, allCategories}) {
```
Comment (jamescodeorg) [inline apps/src/lib/levelbuilder/code-docs-editor/AllCodeDocs.jsx:12]: (minor) If you end up doing another rev, a header comment describing this component might be helpful.  (No need to do this if there are no other changes.)

**Review by jamescodeorg — APPROVED** [review]
Looks good!

File: apps/src/lib/levelbuilder/code-docs-editor/AllCodeDocs.jsx [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+            href="/programming_expressions/new"
+            icon={<i className="fa fa-plus-circle" aria-hidden="true" />}
+            text="New Code Doc"
+            iconBefore
+            openInNewTab
+          />
+        </div>
+      </div>
+      <div>
+        {' '}
```
Comment (tim-dot-org) [inline apps/src/lib/levelbuilder/code-docs-editor/AllCodeDocs.jsx]: is this extra space important, or did the autoformat add it?

**Review by tim-dot-org — APPROVED** [review]

File: apps/src/lib/levelbuilder/code-docs-editor/AllCodeDocs.jsx (hunk 2 above)
Comment (bethanyaconnor) [reply apps/src/lib/levelbuilder/code-docs-editor/AllCodeDocs.jsx]: I didn't add it so it must've been autoformat. But I've seen this happen when I've got a bug then I forget to remove it, so I'll try to remove it.

**Pull request merged**